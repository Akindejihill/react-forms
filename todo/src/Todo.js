import React from 'react';

function Todo (props) {
  return (
    <div>
      {props.todo.task}
      <button onClick={() => props.removeTodo(props.index)}>X</button>
    </div>
  );
}

export default Todo;