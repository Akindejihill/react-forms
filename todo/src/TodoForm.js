import React, { useState } from 'react';

function TodoForm(props) {
  const [task, setTask] = useState('');

  const handleSubmit = event => {
    event.preventDefault();
    props.addTodo(task);
    setTask('');
  };

  return (
    <form onSubmit={handleSubmit}>
      <input
        type="text"
        value={task}
        onChange={event => setTask(event.target.value)}
      />
      <button type="submit">Add Todo</button>
    </form>
  );
}

export default TodoForm;