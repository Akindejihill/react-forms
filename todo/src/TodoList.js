import React, { useState } from 'react';
import Todo from './Todo';
import TodoForm from './TodoForm';

function TodoList() {
  const [todos, setTodos] = useState([]);

  //add todo item to 'todos' state
  const addTodo = (task) => {
    setTodos([...todos, { task }]);
  };

  //splice out the todo item identified by index
  const removeTodo = (index) => {
    const newTodos = [...todos];
    newTodos.splice(index, 1);
    setTodos(newTodos);
  };

  return (
    <div>
      <TodoForm addTodo={addTodo} />
      {todos.map((todo, index) => (
        <Todo key={index} index={index} todo={todo} removeTodo={removeTodo} />
      ))}
    </div>
  );
}

export default TodoList;