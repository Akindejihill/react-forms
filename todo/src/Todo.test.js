import React from "react";
import { render, fireEvent } from "@testing-library/react";
import TodoForm from "./TodoForm";
import TodoList from "./TodoList";
import Todo from "./Todo";

function fakeFunction(){
    console.log("");
}

//smoketests
it("renders without crashing", function () {
    render(<TodoForm />);
});

it("renders without crashing", function () {
    render(<TodoList />);
});

it("renders without crashing", function () {
    render(<Todo key={99} index={99} todo={"test todo list"} removeTodo={fakeFunction} />);
});

//snapshot tests
it("matches snapshot", function () {
    const { asFragment } = render(<TodoForm />);
    expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot", function () {
    const { asFragment } = render(<TodoList />);
    expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot", function () {
    const { asFragment } = render(<Todo key={99} index={99} todo={"test todo list"} removeTodo={fakeFunction} />);
    expect(asFragment()).toMatchSnapshot();
});

it("runs the add todo function on form submit", function () {
    const createMock = jest.fn();
    const { getByText } = render(<TodoForm addTodo={createMock} />);
    const createButton = getByText("Add Todo");
    fireEvent.click(createButton);
    expect(createMock).toHaveBeenCalled();
});
