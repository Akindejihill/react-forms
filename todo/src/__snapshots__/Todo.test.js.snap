// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <input
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      Add Todo
    </button>
  </form>
</DocumentFragment>
`;

exports[`matches snapshot 2`] = `
<DocumentFragment>
  <div>
    <form>
      <input
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        Add Todo
      </button>
    </form>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot 3`] = `
<DocumentFragment>
  <div>
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;
